src/ros_tcp_endpoint/client.py

```diff
@@ -102,7 +102,7 @@ def read_message(conn):
 
             data += packet
 
-        if not data:
+        if full_message_size > 0 and not data:
```

Comment:
Was this necessary because your service does not have a Request message type?

Reply:
My service's request type is empty, yes. And there are plenty of preexisting services like that, so we should handle them :-)